src/utils.js

```diff
@@ -9,8 +10,59 @@ var PLUGIN_NAME = require('../package.json').name;
 
 var urlRegex = /^(https?|webpack(-[^:]+)?):\/\//;
 
+var debug = require('debug-fabulous')()(PLUGIN_NAME + ':utils');
+
+var sourceMapUrlRegEx = /\/\/\# sourceMappingURL\=.*/g
+
+
+var getCommentFormatter = function (file) {
+  var extension = file.relative.split('.').pop(),
+    fileContents =  file.contents.toString(),
+    newline =  detectNewline.graceful(fileContents || ''),
+    commentFormatter = function(url) {
+      return '';
+    };
+
+  if (file.sourceMap.preExisting){
+    debug('preExisting commentFormatter');
+    commentFormatter = function(url) {
+      return file.sourceMap.preExisting;
+    };
+    return commentFormatter
+  }
+
+  switch (extension) {
+    case 'css':
+      debug('css commentFormatter');
+      commentFormatter = function(url) {
+        return newline + "/*# sourceMappingURL=" + url + " */" + newline;
+      };
+      break;
+    case 'js':
+      debug('js commentFormatter');
+      commentFormatter = function(url) {
+        return newline + "//# sourceMappingURL=" + url + newline;
+      };
+      break;
+    default:
+      debug('unknown commentFormatter')
+  }
+
+  return commentFormatter;
+}
+
+var getPreExisting = function(fileContent){
+  if(sourceMapUrlRegEx.test(fileContent)){
```

Comment:
So reusing the same ref for a regex is a major bug in node `0.10` apparently. See http://stackoverflow.com/questions/10229144/bug-with-regexp-in-javascript-when-do-global-search

So I either need to use a new instance of a regex everywhere to make it compatible for all nodes 0.10.X +  or drop support for 0.10.X . For now we will continue to support it I guess. But this sucks.